validator/embedded/src/main/java/org/ethereum/beacon/validator/BeaconChainProposer.java

```diff
@@ -0,0 +1,90 @@
+package org.ethereum.beacon.validator;
+
+import org.ethereum.beacon.chain.observer.ObservableBeaconState;
+import org.ethereum.beacon.consensus.BeaconChainSpec;
+import org.ethereum.beacon.core.BeaconBlock;
+import org.ethereum.beacon.core.BeaconState;
+import org.ethereum.beacon.core.types.BLSSignature;
+import org.ethereum.beacon.core.types.SlotNumber;
+import org.ethereum.beacon.validator.crypto.MessageSigner;
+import tech.pegasys.artemis.ethereum.core.Hash32;
+import tech.pegasys.artemis.util.bytes.Bytes32;
+import tech.pegasys.artemis.util.uint.UInt64;
+
+import static org.ethereum.beacon.core.spec.SignatureDomains.BEACON_PROPOSER;
+import static org.ethereum.beacon.core.spec.SignatureDomains.RANDAO;
+
+/**
+ * An interface of beacon chain proposer. A part of beacon validator logic.
+ *
+ * @see ValidatorService
+ */
+public interface BeaconChainProposer {
+
+  /**
+   * Signs off on a block.
+   *
+   * @param state state at the slot of created block.
+   * @param block block
+   * @param signer message signer.
+   * @param spec Beacon chain spec
+   * @return signature of proposal signed data.
+   */
+  static BLSSignature getProposalSignature(
```

Comment:
Same here. Maybe there is a time to break it down into several interfaces? This change makes significant sense for segregated validator instance.